logstash-core/lib/logstash/agent.rb

```diff
@@ -10,80 +10,130 @@
 LogStash::Environment.load_locale!
 
 class LogStash::Agent
-  attr_reader :logger
+  attr_reader :logger, :state
 
-  def initialize(logger)
-    @logger = logger
+  def initialize(params)
```

Comment:
I'm not a huge fan either, but I think its appropriate for this stage. Once we have more use cases we can determine if another approach makes more sense.